boards/PDM/Inc/Io/Io_Efuse.h

```diff
@@ -0,0 +1,236 @@
+    enum Efuse_Status *status,
+    struct Efuse *     e_fuse);
+
+/**
+ * Get the value of the Aux 1 Fault Register over SPI
+ * @note Reading from the fault register reset's non-latchable faults
+ * @param fault Pointer to the Aux 1 fault status
+ * @param e_fuse Pointer to the Aux1/Aux2 Efuse structure being checked
+ * @return EXIT_CODE_OK if the read was successful
+ *         EXIT_CODE_TIMEOUT if the SPI read timed-out
+ */
+ExitCode Io_Efuse_GetAux1Faults(enum Efuse_Fault *fault, struct Efuse *e_fuse);
+
+/**
+ * Get the value of the Aux 2 Fault Register over SPI
+ * @note Reading from the fault register reset's non-latchable faults
+ * @param fault Pointer to the Aux 2 fault status
+ * @param e_fuse Pointer to the Aux1/Aux2 Efuse structure being checked
+ * @return EXIT_CODE_OK if the read was successful
+ *         EXIT_CODE_TIMEOUT if the SPI read timed-out
+ */
+ExitCode Io_Efuse_GetAux2Faults(enum Efuse_Fault *fault, struct Efuse *e_fuse);
+
+/**
+ * Check if the Aux1/Aux2 efuse is in fault mode
+ * @return true if efuse is in fault mode, else false
+ */
+bool Io_Efuse_Aux1Aux2IsInFaultMode(void);
+
+/**
+ * Check if the Aux1/Aux2 efuse is in fail-safe mode
+ * @return true if efuse is in fail-safe mode, else false
+ */
+bool Io_Efuse_Aux1Aux2IsInFailSafeMode(void);
+
+/**
+ * De-latches the Aux1/Aux2 efuse's latch-able faults
+ * @note Non-latchable faults are reset by reading from the fault registers
+ */
+void Io_Efuse_Aux1Aux2DelatchFaults(void);
+
+/**
+ * Get the current readings for the Aux 1 channel
+ * @param e_fuse Pointer to the Aux1/Aux2 Efuse structure being checked
+ * @param aux1_current Pointer to the Aux 1 current
+ * @return true if the current was measured, else false
+ */
+bool Io_Efuse_GetAux1Current(struct Efuse *e_fuse, float *aux1_current);
+
+/**
+ * Get the current readings for the Aux 2 channel
+ * @param e_fuse Pointer to the Aux1/Aux2 Efuse structure being checked
+ * @param aux1_current Pointer to the Aux 2 current
+ * @return true if the current was measured, else false
+ */
+bool Io_Efuse_GetAux2Current(struct Efuse *e_fuse, float *aux2_current);
+
+/**
+ * Sets current/temperature monitoring option of CSNS pin
+ * @param selection The desired monitoring function which can be selected by
+ * specifying the following values:
+ *                  CSNS_FUNCTION_DISABLED - Current sensing disabled
+ *                  AUX1_CURRENT_SENSE_CHANNEL - Current sensing for Aux 1
+ *                  AUX2_CURRENT_SENSE_CHANNEL - Current sensing for Aux 2
+ *                  CSNS_FUNCTION_TEMPERATURE - Efuse temperature sensing
+ *                  CSNS_FUNCTION_CURRENT_SUM - Current sensing for summed
+ * channels (for Parallel mode)
+ * @param e_fuse Pointer to Aux1/Aux2 Efuse structure being configured
+ * @return EXIT_CODE_OK if the channel configuration was successful
+ *         EXIT_CODE_TIMEOUT if one of the SPI writes timed-out
+ */
+static ExitCode Io_Efuse_Aux1Aux2ConfigureChannelMonitoring(
```

Comment:
I don't think the static function prototypes should be here as we don't want to expose static functions